book/generated-snippets/ts/index.snippet.message-ext-submit.ts

```diff
@@ -0,0 +1,21 @@
+app.on('message.ext.submit', async ({ send, activity }) => {
+  const { commandId } = activity.value;
+  let card: Card;
+
+  if (commandId === 'createCard') {
+    // activity.value.commandContext == "compose"
+    card = createCard(activity.value.data);
+  } else if (commandId === 'getMessageDetails' && activity.value.messagePayload) {
+    // activity.value.commandContext == "message"
+    card = createMessageDetailsCard(activity.value.messagePayload);
+  } else {
```

Comment:
or this?

Reply:
see: https://github.com/microsoft/teams-ai/pull/2418#discussion_r2067420899